File: src/WebUI/libs/auth/shared/src/lib/users/friend-request.response.ts
export const FRIEND_REQUEST_EVENT = {
	SENT: 'FRIEND_REQUEST_SENT',
	ACCEPTED: 'FRIEND_REQUEST_ACCEPTED',
	REJECTED: 'FRIEND_REQUEST_REJECTED',
} as const

export type FriendRequestEvent = (typeof FRIEND_REQUEST_EVENT)[keyof typeof FRIEND_REQUEST_EVENT]

export type FriendRequestResponse = {
	fromAppUserId: string
	fromAppUserUsername: string
	event: FriendRequestEvent
}

File: src/Services/Identity/Identity.Contracts/Responses/Friends/FriendRequestResponse.cs
using Identity.Domain;

namespace Identity.Contracts.Responses.Friends;

public class FriendRequestResponse
{
    public FriendRequestResponse(Guid fromAppUserId, string fromAppUserUsername, string @event)
    {
        FromAppUserId = fromAppUserId;
        FromAppUserUsername = fromAppUserUsername;
        Event = @event;
    }

    public Guid FromAppUserId { get; set; }
    public string FromAppUserUsername { get; set; }
    public string Event { get; set; }

    public static class Status
    {
        public const string Received = "FRIEND_REQUEST_RECEIVED";
        public const string Accepted = "FRIEND_REQUEST_ACCEPTED";
        public const string Rejected = "FRIEND_REQUEST_REJECTED";
    }
}
